app/models/prefill_params.rb

```diff
@@ -0,0 +1,94 @@
+class PrefillParams
+  def initialize(dossier, params)
+    @dossier = dossier
+    @params = params
+  end
+
+  def to_a
+    build_prefill_values.filter(&:prefillable?).map(&:to_h)
+  end
+
+  private
+
+  def build_prefill_values
+    # Builds: [{ stable_id: 1, value: "a value" }, { stable_id: 2, value: "another value"}]
+    stable_ids_and_values =
+      @params
+        .to_unsafe_hash
+        .map { |typed_id, value| { stable_id: stable_id_from_typed_id(typed_id), value: value } }
+        .filter { _1[:stable_id].present? && _1[:value].present? }
+
+    @dossier
+      .find_champs_by_stable_ids(stable_ids_and_values.map { _1[:stable_id] })
+      .map do |champ|
+        value = stable_ids_and_values.find { _1[:stable_id] == champ.stable_id }[:value]
+
+        PrefillValue.new(champ: champ, value: value)
+      end
+  end
+
+  def stable_id_from_typed_id(typed_id)
+    Champ.id_from_typed_id(typed_id).to_i
+  rescue
+    nil
+  end
```

Comment:
Alors, ça c'est indépendant de la pr, mais c'est hyper confusant.

@tchak on peut pas renommer la méthode en `stable_id_from_typed_id` ? je me suis perdu dans le code graphql.

```suggestion
  def stable_id_from_typed_id(typed_id)
    Champ.stable_id_from_typed_id(typed_id).to_i
  rescue
    nil
  end
```

Reply:
Oui. Je le ferai dans une autre PR.